ravenframework/SupervisedLearning/ScikitLearn/GaussianProcess/GPKernels.py

```diff
@@ -0,0 +1,421 @@
+
+#External Modules
+from math import sqrt
+import numpy as np
+from sklearn.gaussian_process.kernels import Kernel as sk_Kernel
+from sklearn.gaussian_process.kernels import ConstantKernel as sk_ConstantKernel
+from sklearn.gaussian_process.kernels import DotProduct as sk_DotProduct
+from sklearn.gaussian_process.kernels import Exponentiation as sk_Exponentiation
+from sklearn.gaussian_process.kernels import ExpSineSquared as sk_ExpSineSquared
+from sklearn.gaussian_process.kernels import Matern as sk_Matern
+from sklearn.gaussian_process.kernels import Product as sk_Product
+from sklearn.gaussian_process.kernels import RationalQuadratic as sk_RationalQuadratic
+from sklearn.gaussian_process.kernels import RBF as sk_RBF
+from sklearn.gaussian_process.kernels import Sum as sk_Sum
+from sklearn.gaussian_process.kernels import WhiteKernel as sk_WhiteKernel
+#External Modules
+
+class Kernel(sk_Kernel):
+  """
+    Base class for skopt.gaussian_process kernels.
+    Supports computation of the gradient of the kernel with respect to X
+  """
+  def __add__(self, b):
+    """
+      Overloads addition operator for kernel objects, to apply correct operation
+      @ In, b, typically a kernel object, element to apply addition to
+      @ Out, resultant kernel object from applying addition
+    """
+    if not isinstance(b, Kernel):
+      return Sum(self, ConstantKernel(b))
+    return Sum(self, b)
+
+  def __radd__(self, b):
+    """
+      Overloads series addition operator for kernel objects, to apply correct operation
+      @ In, b, typically a kernel object, element to apply series addition to
+      @ Out, resultant kernel object from applying series addition
+    """
+    if not isinstance(b, Kernel):
+      return Sum(ConstantKernel(b), self)
+    return Sum(b, self)
+
+  def __mul__(self, b):
+    """
+      Overloads multiplication operator for kernel objects, to apply correct operation
+      @ In, b, typically a kernel object, element to apply multiplication to
+      @ Out, resultant kernel object from applying multiplication
+    """
+    if not isinstance(b, Kernel):
+      return Product(self, ConstantKernel(b))
+    return Product(self, b)
+
+  def __rmul__(self, b):
+    """
+      Overloads series multiplication operator for kernel objects, to apply correct operation
+      @ In, b, typically a kernel object, element to apply series multiplication to
+      @ Out, resultant kernel object from applying series multiplication
+    """
+    if not isinstance(b, Kernel):
+      return Product(ConstantKernel(b), self)
+    return Product(b, self)
+
+  def __pow__(self, b):
+    """
+      Overloads exponentiation operator for kernel objects, to apply correct operation
+      @ In, b, typically a kernel object, element to apply exponentiation to
+      @ Out, resultant kernel object from applying exponentiation
+    """
+    return Exponentiation(self, b)
+
+  def gradient_x(self, x, X_train):
```

Comment:
Is this a carryover from the SciKitLearn API? @wangcj05 what's our policy on snake case method names when replicating an external library?